packages/interaction-kit/src/interactions/autocomplete/autocomplete-interaction.ts

```diff
@@ -0,0 +1,83 @@
+import Application from "../../application";
+import { Autocomplete, RequestBody, ResponseHandler } from "../../interfaces";
+import { AutocompleteInteractionTypes } from "./types";
+import {
+	APIApplicationCommandAutocompleteInteraction,
+	APIApplicationCommandAutocompleteResponse,
+	APIApplicationCommandInteractionDataOption,
+	APIApplicationCommandOptionChoice,
+	APIInteractionGuildMember,
+} from "discord-api-types/v9";
+import { ResponseStatus } from "../../requests/response";
+import type { Snowflake } from "../../structures/snowflake";
+import { Choices } from "../../commands/options/choices";
+import {
+	APIApplicationCommandInteractionDataAutocompleteOption,
+	isAutocompleteOption,
+} from "../../commands/options/option";
+
+// TODO: Specify string or number?
+export default class AutocompleteInteraction<
+	T extends APIApplicationCommandOptionChoice
+> implements Autocomplete<T>
+{
+	public readonly name: string;
+	public readonly token: string;
+	// TODO: Should this be called "focused"?
+	public readonly target: APIApplicationCommandInteractionDataAutocompleteOption;
+
+	public readonly respond: ResponseHandler<APIApplicationCommandAutocompleteResponse>;
+	public readonly options: Map<
+		string,
+		APIApplicationCommandInteractionDataOption
+	> = new Map();
+
+	// TODO: Convert these into Records
+	public readonly channelID: Snowflake | undefined;
+	public readonly guildID: Snowflake | undefined;
+	public readonly member: APIInteractionGuildMember | undefined;
+
+	readonly #application: Application;
+	readonly #type: AutocompleteInteractionTypes;
+
+	constructor(
+		application: Application,
+		// This will need to be changed to a more generic version someday, but for now
+		// is the only type
+		json: RequestBody<APIApplicationCommandAutocompleteInteraction>,
+		respond: ResponseHandler,
+		type: AutocompleteInteractionTypes
+	) {
+		this.#type = type;
+		this.#application = application;
+		this.respond = respond;
+		this.token = json.token;
```

Comment:
Why not private field this too?

Reply:
I think as a whole I need to go through and figure out what fields to expose here and make sure all the data is available - gonna make a new issue for that though